### ISSUE TITLE
CA1862 should not be triggered when an explicit `CultureInfo` is provided.

### ISSUE THREAD

**Issue by m-gallesio** [dotnet/roslyn-analyzers#7672]
### Analyzer

**Diagnostic ID**: [CA1862](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1862): `Use the 'StringComparison' method overloads to perform case-insensitive string comparisons`

### Describe the improvement

The analyzer is triggered by the following code:
```csharp
return LoadAllEnabled().Where(item => item.Text.ToLower(CurrentCulture).Contains(text)).ToList();
```

I think the very fact I am passing a `CultureInfo` explicitly to `ToLower` should be enough to tell that the actual culture *is* potentially meaningful for the comparison.

Moreover, the in-editor description of the rule feels verbose and very lacking in self-confidence (*"keep in mind this might cause subtle changes in behavior, so make sure to conduct thorough testing"*) for an analysis with an automatic code fix.

### Additional context

Internally `Contains` leverages `IndexOf`, which internally uses `CultureInfo.CompareInfo`:

https://github.com/dotnet/runtime/blob/2d5f1f0165e826ac01745df9f3a390905e282ed3/src/libraries/System.Private.CoreLib/src/System/String.Searching.cs#L55

There could be value in exposing variants of `IndexOf`, `Contains` and similar methods which accept a `CultureInfo` or a `CompareInfo` object.

**Comment by bzd3y** [comment]
But are there cultures that would change how cases are compared (especially with a case insensitive comparison)? Maybe there are, but I can't think of how that would work. There are obviously unicameral cultures.

But for those, there is no lowercase/uppercase distinction and the same character should be equal, right?

For any bicameral culture, uppercase and lowercase are different. For a case sensitive comparison they should not be equal, but for a case insensitive comparison they should be. If they aren't equal, then they are not really the same character.

Are you referring to ligature characters? like 'Œ'/'œ' transliterated to "OE"/"oe" or something along those lines? I don't think it can do that for you. You can use `string.Normalize` in some situations, but not where actual transliteration is required.
